src/app/pages/swap/swap-container.tsx

```diff
@@ -1,66 +1,128 @@
 
 import { SwapContainerLayout } from './components/swap-container.layout';
 import { SwapForm } from './components/swap-form';
 import { SwapAsset, SwapFormValues } from './hooks/use-swap';
-import { SwapContext, SwapProvider } from './swap.context';
-
-// TODO: Remove and set to initial state to 0 with live data
-const tempExchangeRate = 0.5;
+import { SwapContext, SwapProvider, SwapSubmissionData } from './swap.context';
 
 export function SwapContainer() {
-  const [exchangeRate, setExchangeRate] = useState(tempExchangeRate);
-  const [isSendingMax, setIsSendingMax] = useState(false);
+  const alexSDK = useState(() => new AlexSDK())[0];
+  const [supportedCurrencies, setSupportedCurrencies] = useState<TokenInfo[]>([]);
+
+  useEffect(() => {
+    alexSDK.fetchTokenList().then(tokenList => {
+      setSupportedCurrencies(tokenList.filter(t => t.availableInSwap));
+    });
+  }, []);
+
   const navigate = useNavigate();
-  const { address } = useCurrentAccountNativeSegwitIndexZeroSigner();
-  const { balance: btcBalance } = useNativeSegwitBalance(address);
-  // TODO: Filter these assets for list to swap, not sure if need?
-  // const allTransferableCryptoAssetBalances = useAllTransferableCryptoAssetBalances();
 
-  // TODO: Replace with live asset list
-  const tempSwapAssetFrom: SwapAsset = {
-    balance: btcBalance,
-    icon: BtcIcon,
-    name: 'Bitcoin',
-  };
+  const allTransferableCryptoAssetBalances = useAllTransferableCryptoAssetBalances();
+  const getAssetFromAlexCurrency = useCallback(
+    (tokenInfo: TokenInfo): SwapAsset => {
+      const currency = tokenInfo.id as Currency;
+      if (currency === Currency.STX) {
+        const balance = allTransferableCryptoAssetBalances.find(
+          x => x.type === 'crypto-currency' && x.blockchain === 'stacks' && x.asset.symbol === 'STX'
+        )!.balance;
+        return { currency, icon: tokenInfo.icon, name: tokenInfo.name, balance };
+      }
+      const balance = allTransferableCryptoAssetBalances.find(
+        x => x.type === 'fungible-token' && alexSDK.getAddressFrom(currency) === x.asset.contractId
+      )?.balance;
+      return {
+        currency,
+        icon: tokenInfo.icon,
+        name: tokenInfo.name,
+        balance: balance ?? createMoney(0, tokenInfo.name, tokenInfo.decimals),
+      };
+    },
+    [allTransferableCryptoAssetBalances]
+  );
 
-  const tempSwapAssetTo: SwapAsset = {
-    balance: createMoney(new BigNumber(0), 'xBTC', 0),
-    icon: XBtcIcon,
-    name: 'Wrapped Bitcoin',
-  };
+  const swappableAssets: SwapAsset[] = useMemo(
+    () => supportedCurrencies.map(getAssetFromAlexCurrency),
+    [getAssetFromAlexCurrency, supportedCurrencies]
+  );
 
-  function onSubmitSwapForReview(values: SwapFormValues) {
-    navigate(RouteUrls.SwapReview, {
-      state: { ...values },
+  const [swapSubmissionData, setSwapSubmissionData] = useState<SwapSubmissionData>();
+  const [slippage, _setSlippage] = useState(0.04);
```

Comment:
Currently the UI is missing the ability to update slippage, we have it default to 4% atm.